PoSSapi/src/Application/Orders/Queries/GetAllOrdersQuery.cs

```diff
@@ -0,0 +1,60 @@
+using AutoMapper;
+using AutoMapper.QueryableExtensions;
+using MediatR;
+using PoSSapi.Application.Common.Interfaces;
+using PoSSapi.Application.Common.Mappings;
+using PoSSapi.Application.Common.Models;
+using PoSSapi.Domain.Entities;
+using PoSSapi.Domain.Enums;
+
+
+namespace PoSSapi.Application.Orders.Queries;
+
+public record GetAllOrdersQuery : IRequest<PaginatedList<Order>>
+{
+    public int PageNumber { get; init; } = 1;
+    public int PageSize { get; init; } = 10;
+    public Guid? UserId { get; init; }
+    public OrderStatus? Status { get; init; }
+}
+
+public class GetAllOrdersQueryHandler : IRequestHandler<GetAllOrdersQuery, PaginatedList<Order>>
+{
+    private readonly IApplicationDbContext _context;
+    private readonly IMapper _mapper;
+
+    public GetAllOrdersQueryHandler(IApplicationDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    public async Task<PaginatedList<Order>> Handle(GetAllOrdersQuery request, CancellationToken cancellationToken)
+    {
+        IQueryable<Order> orders;
+
+        if (request.UserId != null && request.UserId != null)
```

Comment:
I'd think one check is enough in this if. Anyway, i'd suggest just using two ifs instead of 4 that you have here like:
`orders = _context.Orders
if(...)
{
orders = orders.where(onething)
}
if(...)
{
orders = orders.where(otherthing)
}
`